### ISSUE TITLE
Accessibility: Dialog contents is not accessible with VoiceOver

### ISSUE THREAD

**Issue by MarcoZehe** [electron/electron#22928]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * 7.1.11
* **Operating System:** MacOS 10.15.4
  * MacOS 10.15.4
* **Last Known Working Electron version:**
  * unknown

### Expected Behavior

When opening a dialog, such as About, VoiceOver should be able to read and interact with it and its controls.

### Actual Behavior

VoiceOver does not see dialog contents in many cases.

### To Reproduce

1. Download and install Visual Studio code.
2. Start VoiceOver, then Code.
3. Go to Visual Studio Code/About, and press Enter.
4. Try interacting with the dialog and read the contents, or navigate to the OK and Copy buttons using VoiceOver commands.

### Additional Information

VoiceOver will track the focus somewhat, but not reliably, when using the Tab key to navigate. But VoiceOver commands such as CTRL+Option+Arrow keys don't work, VoiceOver only sees the Close, Minimize, and Zoom buttons.

This looks similar to #17207, only that this pertains to dialogs, not web content.

**Comment by cookiecrook** [comment]
@kloots FYI

**Comment by codebytere** [comment]
@MarcoZehe hmm - i can reproduce this with VSCode, but cannot seem to reproduce it on Electron itself. I tested with [this fiddle](https://gist.github.com/cdce8019f452cc29721cad6836a2b8bc), and was successfully able to navigate the content with VoiceOver and the arrow keys/tab, as seen here:

![dialog_accessibility](https://user-images.githubusercontent.com/2036040/78629098-f285f600-784a-11ea-8630-b110c0e2142f.gif)

Would it be possible for you to reproduce this in a standalone fiddle, which is not app-specific? If not, it may perhaps be more effective to report this to VSCode itself, as they may have configured something to interfere with this accidentally 🤔

**Comment by MarcoZehe** [comment]
@codebytere Thank you for that Gist! I have never ran any Electron build or such myself, so will need some time to get into how this all works and then hopefully be able to run your fiddle. In the meantime, VS Code is not the only app I am seeing this in. Slack, which is not open-source, is another one. Perhaps @klots can speak to how the Slack app opens dialogs like the About dialog or the Check for Updates one if there's no update. On Windows, these all work, but on Mac, they don't. They grab focus, but the accessibility objects aren't visible to VoiceOver. Instead, VoiceOver thinks focus has returned to the element that was active before the menu that invoked the dialog had been opened.

**Comment by MarcoZehe** [comment]
Sorry, still haven't got around to diving into how to set up an Electron dev environment and get that fiddle to run. In the meantime, @deepak1556, any idea on @codebytere 's question from above re what VS Code does with displaying those dialogs?

**Comment by isidorn** [comment]
Here's a vscode issue https://github.com/microsoft/vscode/issues/55585
And this issue seems like a duplicate of this one which I filled a long time ago https://github.com/electron/electron/issues/14234

@codebytere you have all the details in my original issue. Also @MarcoZehe is a screen reader users so a descipriton might be better than an animated gif.

**Comment by MarcoZehe** [comment]
Thanks @isidorn , closing my issue as a duplicate of your issue #14234.
